microconfig-core/src/main/java/io/microconfig/core/properties/PropertyImpl.java

```diff
@@ -33,15 +35,28 @@ public static Property parse(String keyValue, ConfigFormat configFormat, Declari
         String key = keyValue.substring(temp ? TEMP_VALUE.length() : 0, separatorIndex).trim();
         String value = keyValue.substring(separatorIndex + 1).trim();
 
-        return new PropertyImpl(key, value, temp, configFormat, source);
+        return new PropertyImpl(key, value, temp, configFormat, source, null);
     }
 
     public static Property property(String key, String value, ConfigFormat configFormat, DeclaringComponent source) {
-        return new PropertyImpl(key, value, false, configFormat, source);
+        return new PropertyImpl(key, value, false, configFormat, source, null);
+    }
+
+    public static Property atProperty(String key, String value, ConfigFormat configFormat, DeclaringComponent source) {
```

Comment:
maybe rename to envProperty

Reply:
renamed and moved to EnvProperty decorator